sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/monitor/applicationinsights/spring/selfdiagnostics/SelfDiagAutoConfig.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.applicationinsights.spring.selfdiagnostics;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+/**
+ Main configuration entry point of the self-diagnostics
+ **/
+@Configuration(proxyBeanMethods = false)
+@Import({DefaultLogConfig.class, LogbackSelfDiagConfig.class, JdbcSelfDiagConfig.class})
+public class SelfDiagAutoConfig {
+    private static final Logger LOG = LoggerFactory.getLogger(SelfDiagAutoConfig.class);
+    private static final String SELF_DIAGNOSTICS_LEVEL_ENV_VAR = "APPLICATIONINSIGHTS_SELF_DIAGNOSTICS_LEVEL";
+
+    @Bean
+    SelfDiagnosticsLevel selfDiagnosticsLevel() {
+        String selfDiagLevelEnvVar = System.getenv(SELF_DIAGNOSTICS_LEVEL_ENV_VAR);
+        if (selfDiagLevelEnvVar == null) {
+            return SelfDiagnosticsLevel.INFO;
+        }
+        try {
+            return SelfDiagnosticsLevel.valueOf(selfDiagLevelEnvVar);
+        } catch (IllegalArgumentException e) {
+            LOG.warn("Unable to find the self-diagnostics level related to " + selfDiagLevelEnvVar + "defined with " + SELF_DIAGNOSTICS_LEVEL_ENV_VAR + " environment variable.", e);
+            return SelfDiagnosticsLevel.INFO;
+        }
+    }
+    @Bean
+    OtelSelfDiag otelSelfDiag(ApplicationContext applicationContext, Logger selfDiagnosticsLogger) {
+        return new OtelSelfDiag(applicationContext, selfDiagnosticsLogger);
+    }
+
+    @Bean
+    ExecutionEnvSelfDiag executionEnvSelfDiag(Logger selfDiagnosticsLogger) {
+        return new ExecutionEnvSelfDiag(selfDiagnosticsLogger);
+    }
```

Comment:
can `JdbcSelfDiag` be defined here also for consistency?

Reply:
No, because `JdbcSelfDiag` has only to be created in case of a `java.sql.DataSource` (see `JdbcSelfDiagConfig`).